src/macaron/malware_analyzer/pypi_heuristics/sourcecode/suspicious_setup.py

```diff
@@ -123,3 +124,88 @@ def analyze(self, api_client: PyPIRegistry) -> tuple[HeuristicResult, dict]:
         if suspicious_setup:
             return HeuristicResult.FAIL, {"import_module": imports}
         return HeuristicResult.PASS, {"import_module": imports}
+
+    def extract_from_ast(self, source_content: str) -> set[str]:
+        """Extract imports from source code using the parsed AST.
+
+        Parameters
+        ----------
+        source_content: str
+            The source code as a string.
+
+        Returns
+        -------
+        set[str]
+            The set of imports.
+
+        Raises
+        ------
+        SyntaxError
+            If the code could not be parsed.
+        """
+        imports = set()
+        tree = ast.parse(source_content)
+        for node in ast.walk(tree):
+            if isinstance(node, ast.Import):
+                for alias in node.names:
+                    imports.add(alias.name)
+            elif isinstance(node, ast.ImportFrom):
+                module = node.module
+                if node.level != 0 or not module:
+                    # Relative imports are ignored.
+                    continue
+                imports.add(module)
+
+        return imports
```

Comment:
I made some comparison between this newer implementation and the old one. I noticed that there are some differences between them.
```
>>> line = "from .a.b.c import x, y, z"
>>> old(line)
{'.a.b.c.y', '.a.b.c.x', '.a.b.c.z'}
>>> new(line)
set()

>>> line =  "from a import x,y,z"
>>> old(line)
set()
>>> new(line)
{'a'}

>>> line = "from a.b.c import x,y,z"
>>> old(line)
set()
>>> new(line)
{'a.b.c'} 
```
Is it okay to say that these cases are as expected as we completely ignore any relative imports and only takes the absolute ones?
Hmm then my next question would be whether the behavior of the old implementation (i.e accept `from .a.b.c import x, y, z` and ignore `from a import x,y,z`) was incorrect, or was it intended? 
@behnazh-w Do you happen to remember what is the intended behavior of this logic :thinking: ?

Reply:
We should support both relative and absolute imports. The idea here is to catch all possible imports. So, if the previous implementation does not handle `from a import x,y,z`, it could be a bug.

Reply:
Oh okay, so then this method should return **all modules** imported regardless of its type.
- Absolute import `from A ...` or `from A.B.C`
- Relative import `from . ...` or `from .A.B.C`